content/blog/air/index.Rmd

```diff
@@ -0,0 +1,349 @@
+
+photo:
+  url: https://unsplash.com/photos/above-cloud-photo-of-blue-skies-yQorCngxzwI
+  author: Taylor Van Riper
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [programming] 
+tags: []
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+We're thrilled to announce [Air](https://posit-dev.github.io/air/), an extremely fast R formatter.
+Formatters are used to automatically style code, but I find that it's much easier to show what Air can do rather than tell, so we'll start with a few examples.
+In the video below, we're inside [Positron](https://positron.posit.co/) and we're looking at some unformatted code.
+Saving the file (yep, that's it!) invokes Air, which automatically and instantaneously formats the code.
+
+<video controls autoplay loop muted width="100%" src="video/case-when.mov" style="border: 2px solid #CCC;">
+
+</video>
+
+Next, let's go over to [RStudio](https://posit.co/products/open-source/rstudio/).
+Here we've got a pipe chain that could use a little formatting.
+Like in Positron, just save the file:
+
+<video controls autoplay loop muted width="100%" src="video/ggplot.mov" style="border: 2px solid #CCC;">
+
+</video>
+
+Lastly, we'll jump back into Positron.
+Rather than formatting a single file on save, you might want to instead format an entire project (particularly when first adopting Air).
+To do so, just run `air format .` in a terminal from the project root, and Air will recursively format any R files it finds along the way (smartly excluding known generated files, like `cpp11.R`).
+Here we'll run Air on dplyr for the first time ever, analyzing and reformatting over 150 files instantly:
+
+<video controls autoplay loop muted width="100%" src="video/project.mov" style="border: 2px solid #CCC;">
+
+</video>
+
+Throughout the rest of this post you'll learn what a formatter is, why you'd want to use one, and you'll learn a little about how Air decides to format your R code.
+
+Note that Air is still in alpha, so there may be some breaking changes over the next few releases, but we still feel very confident in the current state of Air, and can't wait for you to try it!
+
+## Installing Air
+
+If you already know how formatters work and want to jump straight in, follow one of the guides below:
+
+-   For Positron, Air is [available](https://open-vsx.org/extension/posit/air-vscode) on OpenVSX as an Extension.
+    Install it from the Extensions pane within Positron, then read our [Positron guide](https://posit-dev.github.io/air/editor-vscode.html).
+
+-   For VS Code, Air is [available](https://marketplace.visualstudio.com/items?itemName=Posit.air-vscode) on the VS Code Marketplace as an Extension.
+    Install it from the Extensions pane within VS Code, then read our [VS Code guide](https://posit-dev.github.io/air/editor-vscode.html).
+
+-   For RStudio, Air can be set as an external formatter, but you'll need to install the command line tool for Air first.
```

Comment:
Will RStudio switch to an stdin ingestion style? If so should we warn about the upcoming breaking change?